src/agent/multitype_queue/include/imultitype_queue.hpp

```diff
@@ -81,7 +83,7 @@ class IMultiTypeQueue
      * @return std::vector<Message> A vector of messages fetched from the queue.
      */
     virtual std::vector<Message> getNextN(MessageType type,
-                                          int messageQuantity,
+                                          std::variant<const int, const size_t> messageQuantity,
```

Comment:
Like expressed in some other comment, I rather see something like `std::variant<const MessageCount, const MessageBytes>` and have these types prohibit implicit conversions between them and other numeric types.